torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1427,9 +1428,12 @@ struct DumpPipe {
 };
 #endif
 
+std::string ProcessGroupNCCL::createLogPrefix() const {
+  return c10::str("[PG ", uid_, " Rank ", rank_, "] ");
+}
+
 const std::string& ProcessGroupNCCL::logPrefix() const {
-  static std::string prefix = c10::str("[PG ", uid_, " Rank ", rank_, "] ");
```

Comment:
ohh, i see what's going on . the static prefix here is shared across all ProcessGroup instances, meaning whoever calls it first sets the `uid_` and `rank_` and then any other sub-PGs on that process will inherit it from the first caller.  So since we log from PG0 first, its always PG0.  Good catch.

Reply:
yep that's exactly what happened and why we always see PG 0 Rank (global rank) because that PG is always created first and this function is called in the constructor.

Reply:
I borrowed this idea to create a method `globalRank` so that all NCCL PG instances can get its global ranks now.